crates/tabby/src/serve/context.rs

```diff
@@ -0,0 +1,89 @@
+use anyhow::Result;
+use nvml_wrapper::Nvml;
+use sysinfo::{CpuExt, System, SystemExt};
+
+// TabbyContext is the struct for global shared context
+// to be used by router app.
+pub struct TabbyContext {
+    pub cpu_stats_manager: CPUStatsManager,
+    pub gpu_stats_manager: GPUStatsManager,
+}
+
+impl TabbyContext {
+    pub fn new() -> Self {
+        let cpu_stats_manager = CPUStatsManager::init();
+        let gpu_stats_manager = GPUStatsManager::init();
+        Self {
+            cpu_stats_manager,
+            gpu_stats_manager,
+        }
+    }
+}
+
+pub struct CPUStat {
+    pub info: String,
+    pub count: usize,
+}
+pub struct CPUStatsManager {
+    system: System,
+}
+
+impl CPUStatsManager {
+    pub fn init() -> Self {
+        Self {
+            system: System::new_all(),
+        }
+    }
+
+    pub fn get_stats(self: &mut Self) -> CPUStat {
```

Comment:
I suggest create a single function doing the same functionality (returns GPU info string and count), and call this function in health state. Instead of embeding this into a context